api/v1alpha1/mustgather_types.go

```diff
@@ -63,13 +63,13 @@ type MustGatherSpec struct {
 
 // +k8s:openapi-gen=true
 type ProxySpec struct {
-	// httpProxy is the URL of the proxy for HTTP requests.  Empty means unset and will not result in an env var.
-	// +optional
-	HTTPProxy string `json:"httpProxy,omitempty"`
+	// httpProxy is the URL of the proxy for HTTP requests.
+	// +kubebuilder:validation:Required
```

Comment:
need to add `+required` also, need to check the `kube-openapi-gen` version here if it's relatively latest only `+required` will suffice, else we need `+kubebuilder:validation:Required` as well. 

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required